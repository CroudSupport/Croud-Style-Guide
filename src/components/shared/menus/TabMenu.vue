<template>
    <div class="ui menu">
        <div v-for="item in menuItems" class="yellow item" :class="{active: localActive === item.component}" @click="localActive = item.component" v-html="item.name"></div>
    </div>
</template>

<script>
    
    /**
    * Basic tabular menu component
    * @example ./croud-tab-menu.md
    */

    export default {
        name: 'croud-tab-menu',

        model: {
            prop: 'active',
        },

        props: {
            /**
            * State which tab is active. (by name of menu item)
            */
            active: {
                required: true,
            },
            /**
            * Menu items to be displayed on tab menu.
            */
            menuItems: {
                required: true,
                type: Array,
            },
        },

        computed: {
            localActive: {
                get() {
                    return this.active
                },
                set(val) {
                    this.$emit('input', val)
                },
            },
        },
    }
</script>

<style scoped>
    .pointing {
        cursor: pointer;
    }
</style>
