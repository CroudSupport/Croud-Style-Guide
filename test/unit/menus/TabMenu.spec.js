import Vue from 'vue'
import TabMenu from '../../../src/components/shared/menus/TabMenu'

const Constructor = Vue.extend(TabMenu)

const vm = new Constructor({
    propsData: {
        active: 'Journal',
        menuItems: [
            {
                name: 'Planning',
                component: 'PlanningComponent',
            },
            {
                name: 'Journal',
                component: 'JournalComponent',
            },
            {
                name: 'Calander',
                component: 'CalanderComponent',
            },
        ],
    },
}).$mount()

describe('Tab Menu', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })
})
