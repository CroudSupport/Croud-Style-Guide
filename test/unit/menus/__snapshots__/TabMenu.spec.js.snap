// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Menu should match the snapshot 1`] = `
<div
  class="ui menu"
>
  <div
    class="yellow item"
  >
    Planning
  </div>
  <div
    class="yellow item"
  >
    Journal
  </div>
  <div
    class="yellow item"
  >
    Calander
  </div>
</div>
`;
